File: 2023_8/baekjoon/2501.py
import sys

N, K = map(int, sys.stdin.readline().split())
divisor = [1, N]

for i in range(2, int(N**0.5) + 1):
    if N % i == 0:
        divisor.append(i)
        divisor.append(N//i)

divisor = sorted(list(set(divisor)))

print(0 if len(divisor) < K else divisor[K - 1])

File: 2023_8/baekjoon/2501.rs
use std::io::stdin;

fn main() {
    let mut buf = String::new();
    stdin().read_line(&mut buf).unwrap();
    let mut buf = buf.split_whitespace();

    let n = buf.next().unwrap().parse::<u32>().unwrap();
    let k = buf.next().unwrap().parse::<usize>().unwrap();
    let mut divisor: Vec<u32> = vec![1, n];

    for i in 2..((n as f32).sqrt() as u32 + 1) {
        if n % i == 0 {
            divisor.push(i);
            if n / i != i {
                divisor.push(n / i);
            }
        }
    }

    if divisor.len() < k {
        println!("0");
    } else {
        divisor.sort();
        println!("{}", divisor[k - 1]);
    }
    return;
}
